src/core/load-fragment.c

```diff
@@ -808,7 +808,7 @@ int config_parse_exec(
                                 if (!fullpath)
                                         return log_oom();
 
-                                if (access(fullpath, F_OK) >= 0) {
+                                if (access(fullpath, X_OK) >= 0) {
```

Comment:
maybe add a comment, explaining why this is done, because it is really weird

Reply:
This part is removed in a subsequent commit. I made it the change here to avoid changing the test logic together with the later change to use find_executable(). (And also to possibly make it easier to backport this change on its own.)